kore/src/Kore/Step/RewriteStep.hs

```diff
@@ -331,12 +336,17 @@ applyRewriteRulesParallel
     -> simplifier (Results (RulePattern RewritingVariableName))
 applyRewriteRulesParallel
     unificationProcedure
-    (map getRewriteRule -> rules)
-    initial
+    (map (mapRuleVariables resetRuleVariable . getRewriteRule) -> rules)
+    (Pattern.mapVariables resetConfigVariable -> initial)
```

Comment:
This function should not need to alter the variables in the rules or configuration: the rules and configuration should already have the right kind of variables, and renaming every variable would be very slow (that's what we want to avoid).